torch/_decomp/decompositions.py

```diff
@@ -1991,12 +1991,13 @@ def upsample_compute_output_size(input_size, output_size, scale_factors):
             lambda: "Must specify exactly one of output_size and scale_factors",
         )
         utils.check(len(scale_factors) == spatial_dimensions, lambda: "")
-        return [
-            # Returning output_size as float. We cannot convert it to int directly,
-            # as latter computation of scale_factor is relying output size being float
-            sym_float(input_size[i + 2] * scale_factors[i])
-            for i in range(spatial_dimensions)
-        ]
+        output_size = []
+        for i, s in enumerate(scale_factors):
+            if int(s) == s:
+                output_size.append(input_size[i + 2] * int(scale_factors[i]))
```

Comment:
I think it doesn't matter, but this diverges behavior when the input size * scale factor result is sufficiently large, because the floating point calculation can't represent the integral result. But for double that's at 2 ** 53 which is way too big.